react/GalleryLayoutSwitcher.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useCallback } from 'react'
+import classNames from 'classnames'
+import { useCssHandles } from 'vtex.css-handles'
+import { SearchPageContext } from 'vtex.search-page-context'
+
+const { useSearchPageState, useSearchPageStateDispatch } = SearchPageContext
+
+const CSS_HANDLES = ['galleryLayoutSwitcher'] as const
+
+const GalleryLayoutSwitcher: React.FC = ({ children }) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  const {
+    selectedGalleryLayout,
+    focusedGalleryLayout,
+    galleryLayouts,
+  } = useSearchPageState()
+  const dispatch = useSearchPageStateDispatch()
+
+  const gallerySwitcherClasses = classNames(
+    handles.galleryLayoutSwitcher,
+    'flex flex-row bn'
+  )
+
+  const navigateWithKeys = useCallback(
+    (e: React.KeyboardEvent<HTMLDivElement>) => {
+      if (
+        galleryLayouts &&
+        (e.key === 'ArrowLeft' ||
+          e.key === 'ArrowUp' ||
+          e.key === 'ArrowRight' ||
+          e.key === 'ArrowDown')
+      ) {
+        const currentFocusedIndex = galleryLayouts.findIndex(
+          (layout) =>
+            layout.name === (focusedGalleryLayout ?? selectedGalleryLayout)
+        )
+
+        if (currentFocusedIndex === -1) {
+          console.error(
+            `No layout defined with name ${focusedGalleryLayout}. Check if there are unnecessary layout options with this name.`
+          )
+        }
+
+        let newFocusedLayoutIndex = 0
+
+        if (e.key === 'ArrowLeft') {
+          newFocusedLayoutIndex = Math.max(0, currentFocusedIndex - 1)
+        } else if (e.key === 'ArrowRight') {
+          newFocusedLayoutIndex = Math.min(
+            currentFocusedIndex + 1,
+            galleryLayouts.length - 1
+          )
+        } else if (e.key === 'ArrowUp') {
+          newFocusedLayoutIndex =
```

Comment:
What is the difference using arrow up and down from left and right?

Reply:
Left and right keys do nothing when they reach their corners. Up and down loops through the items when that happens.